great_expectations/checkpoint/actions.py

```diff
@@ -328,13 +342,16 @@ def _run(  # type: ignore[override] # signature does not match parent  # noqa: C
             # To send a notification with a link to the validation result, we need to have created the validation  # noqa: E501
             # result in cloud. If the user has configured the store action after the notification action, they will  # noqa: E501
             # get a warning that no link will be provided. See the __init__ method for ActionListValidationOperator.  # noqa: E501
-            if (
-                "store_validation_result" in payload
-                and "validation_result_url" in payload["store_validation_result"]
-            ):
-                validation_result_urls.append(
-                    payload["store_validation_result"]["validation_result_url"]
-                )
+            store_validation_results = action_context.filter_results_by_class(
+                StoreValidationResultAction
+            )
+            if store_validation_results:
+                for payload in store_validation_results:
+                    if "validation_result_url" in payload:
+                        validation_result_urls.append(
+                            payload["store_validation_result"]["validation_result_url"]
+                        )
```

Comment:
This logic is a bit iffy but its going away with our removal of the `StoreValidationResultAction` (not sure how to get parity here just yet though)